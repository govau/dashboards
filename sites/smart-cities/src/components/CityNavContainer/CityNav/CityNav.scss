@import '~style/main';

.link {
  display: block;
  transition: $animation-duration;
  color: $color-white;
  font-size: 16px;
  letter-spacing: -0.3px;

  .citySelected & {
    color: $color-primary--060;
  }

  &:hover {
    background: $color-primary--800;
    color: $color-white;
  }

  &.link__active {
    background: $color-primary--900;
    color: $color-white;
  }
}

.allCitiesLink {
  display: flex;
  align-items: center;
  height: $top-nav-height-mobile;
  margin-bottom: 6px;
  padding: 0 16px;
  background: $color-primary--800;
  font-size: 22px;
  font-weight: 300;
  letter-spacing: 0.5px;

  &:hover {
    background: $color-primary--900;
  }

  .citySelected & {
    color: $color-primary--080;
  }

  &.link__active {
    font-weight: 500;
  }

  @include for-tablet-landscape-up {
    font-size: 24px;
    font-weight: 300;
  }

  @include for-desktop-up {
    height: $top-nav-height-desktop;
  }
}

.cityLink {
  padding: 0 10px 0 30px;
  line-height: 40px; // height is set because when selected this turns bold

  &.link__active {
    font-weight: 500;
  }
}
