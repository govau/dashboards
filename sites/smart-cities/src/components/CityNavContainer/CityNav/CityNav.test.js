import React from 'react';
import { shallow } from 'enzyme';
import CityNav from './CityNav';

jest.mock('../../Pill/Pill', () => 'Pill');

const defaultProps = {
  cityId: 'perth',
  cities: [
    {
      id: 'perth',
      name: 'Perth',
    },
    {
      id: 'sydney',
      name: 'Sydney',
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityNav {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
