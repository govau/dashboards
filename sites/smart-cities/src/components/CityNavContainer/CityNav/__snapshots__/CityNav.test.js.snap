// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"citySelected\\">
  <NavLink className=\\"link allCitiesLink\\" activeClassName=\\"link__active\\" isActive={[Function]} to=\\"/all-cities/overview\\" ariaCurrent=\\"true\\">
    All cities
  </NavLink>
  <NavLink className=\\"link cityLink\\" activeClassName=\\"link__active\\" isActive={[Function]} to=\\"/perth/overview\\" ariaCurrent=\\"true\\">
    Perth
  </NavLink>
  <NavLink className=\\"link cityLink\\" activeClassName=\\"link__active\\" isActive={[Function]} to=\\"/sydney/overview\\" ariaCurrent=\\"true\\">
    Sydney
  </NavLink>
</div>"
`;
